src/Umbraco.Core/ObjectResolution/ManyObjectsResolverBase.cs

```diff
@@ -160,7 +156,7 @@ protected virtual IEnumerable<Type> InstanceTypes
         /// Gets or sets the <see cref="HttpContextBase"/> used to initialize this object, if any.
         /// </summary>
         /// <remarks>If not null, then <c>LifetimeScope</c> will be <c>ObjectLifetimeScope.HttpRequest</c>.</remarks>
-        protected HttpContextBase CurrentHttpContext { get; private set; }
+        protected HttpContextBase CurrentHttpContext { get { return _httpContextGetter == null ? null : _httpContextGetter(); } }

```

Comment:
@zpqrtbnk this getter should do the null check and exception throwing that the ctor used to do:

```
throw new InvalidOperationException("Use alternative constructor accepting a HttpContextBase object in order to set the lifetime scope to HttpRequest when HttpContext.Current is null");
```
because if it's null other things are going to break and we'll get null ref exceptions, in theory this should never be null if the resolver is in fact a request based lifespan.